Sources/Subs.php

```diff
@@ -4982,9 +4983,31 @@ function inet_dtop($bin)
 	if ($db_type == 'postgresql')
```

Comment:
If we pretend that using PostgreSQL somehow guaranteed that `$bin` could never carry incorrectly formatted data, your code would at least need to look like this

```
if ($db_type == 'postgresql')
    return $bin;
else
{
    // code to test whether $bin is a normal string or has been packed into binary format
}